tests/testthat/test-wildcards.R

```diff
@@ -24,5 +24,35 @@ test_that("HTML `data-*` & `aria-* ` wildcards are passed along to layout approp
   expect_equal(x$props$`data-foo`, 1)
 })
 
+
+test_that("restricted pattern matching callback selectors must be formatted as a list", {
+  app <- Dash$new()
+  app$layout(htmlDiv(list(
+    htmlButton("Add Filter", id="add-filter", n_clicks=0),
+    htmlDiv(id="dropdown-container", children=list()),
+    htmlDiv(id="dropdown-container-output")
+  )))
+  
+  expect_error(
+    app$callback(
+      output(id="dropdown-container-output", property="children"),
+      params = list(
+        input(id=list("index" = dash:::ALL, "type" = "filter-dropdown"), property= "value")
+      ),
+      display_output <- function(test){
+        return(htmlDiv(
+          lapply(1:length(test), function(x){
```

Comment:
I generally use `seq_along(test)` instead of `1:length(test)`, for a better discussion of the reasons why than what I would likely provide, you might want to skim this chapter: https://r4ds.had.co.nz/iteration.html.

Reply:
Thanks for pointing it out! Changed all of those to `seq_along` in 577deb4.